lib/spack/spack/test/util/editor.py

```diff
@@ -132,3 +132,17 @@ def assert_exec(exe, args):
         assert args == [vim_exe, '/path/to/file']
 
     ed.editor('/path/to/file', _exec_func=assert_exec)
+
+
+def test_no_editor():
+    if 'VISUAL' in os.environ:
+        del os.environ['VISUAL']
+    if 'EDITOR' in os.environ:
+        del os.environ['EDITOR']
+    os.environ['PATH'] = ''
+
+    def assert_exec(exe, args):
+        assert False
```

Comment:
Why do we have to `assert False` here? Wouldn't:
```python
def _no_op_exec(exe, args):
    pass
```
give a better error message (i.e. not cluttered with a failing assertion?)

Reply:
It asserts to check that exec is never called. I'd forgotten that assert is implemented as an exception, though. I suppose my change to catch `Exception` in `editor.try_exec` will prevent all the tests editor.py from failing... I'll change that back to `OSError`.

Reply:
Yes, but "does not try to exec a program when none exists" and "properly raises an EnvironmentError" are two independent behaviours to verify. For example, if the implementation was missing the return after `try_exec` and the exec did not assert, then this test would pass while the implementation would be broken.